samples/template-operator/README.md

```diff
@@ -34,270 +66,511 @@ Additionally, it hides Kubernetes boilerplate code to develop fast and efficient
+
+2. As part of the controller's `SetupWithManager()` in the Sample CR [controller implementation](controllers/sample_controller.go), we now have to tell the declarative reconciler how to reconcile the object.
+   For this, we need to inject the necessary information about the declarative intention on what to reconcile with `Inject(...)`.
+
+   In its most simple form, the new Setup could look like this (assuming all the settings from above were used):
 
-2. As part of reconciler's `SetupWithManager()` in the Sample CR [controller implementation](./operator/controllers/sample_controller.go), declarative options have been used.
-   Similarly, implement the required options in your controller.
    ```go
-   return r.Inject(mgr, &v1alpha1.Sample{},
-        declarative.WithManifestResolver(manifestResolver),
-        declarative.WithCustomResourceLabels(map[string]string{"sampleKey": "sampleValue"}),
-        declarative.WithPostRenderTransform(transform),
-        declarative.WithResourcesReady(true),
-        declarative.WithFinalizer(sampleFinalizer),
-   )
+    package controllers
+    import (
+	    "github.com/kyma-project/module-manager/operator/pkg/declarative"
+        metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	    "sigs.k8s.io/controller-runtime/pkg/client"
+	    ctrl "sigs.k8s.io/controller-runtime"
+	    "k8s.io/apimachinery/pkg/runtime"
+   
+	    operatorv1alpha1 "github.com/kyma-project/test-operator/operator/api/v1alpha1"
+    )
+   
+    // SampleReconciler reconciles a Sample object
+    type SampleReconciler struct {
+      declarative.ManifestReconciler
+      client.Client
+      Scheme *runtime.Scheme
+    }
+   
+    // SetupWithManager sets up the controller with the Manager.
+    func (r *SampleReconciler) SetupWithManager(mgr ctrl.Manager) error {
+    if err := r.Inject(
+      mgr, &operatorv1alpha1.Sample{},
+      declarative.WithCustomResourceLabels(map[string]string{"sampleKey": "sampleValue"}),
+      declarative.WithResourcesReady(true),
+      declarative.WithFinalizer("sample-finalizer"), 
+    ); err != nil {
+      return err
+    }
+    
+    return ctrl.NewControllerManagedBy(mgr).
+	    For(&operatorv1alpha1.Sample{}).
+	    Complete(r)
+    }
    ```
    These options can be used to modify manifest installation and uninstallation. Some options are applied as a manifest pre-processing step and others as post-processing.
-   More details on these steps can be found in the [options documentation](https://github.com/kyma-project/module-manager/blob/main/operator/pkg/declarative/options.go).
+   More details on these steps can be found in the [options documentation](https://github.com/kyma-project/module-manager/blob/main/operator/pkg/declarative/options.go) or in the reference implementation.
 
 3. A **mandatory** requirement of this reconciler is to provide the option `declarative.WithManifestResolver(manifestResolver)`, as it holds the chart information to be processed by the declarative reconciler.
```

Comment:
That is not 100% correct. They can use `declarative.WithManifestResolver(...)` or `declarative.WithDefaultResolver` - one of them need to be used, but not specifically `WithManifestResolver`.


Since if they do not want to hardcode their InstallationSpec values, we gave them the option to include them into their CR. Thus they do not need to come up with an own resolver and can save code duplication, if they do not have a fancy Resolver which needs to do specials operations.
The default resolver will take care of resoling the spec of the CR into a valid InstallationSpec.
```
declarative.WithDefaultResolver(),
```